addons/web/controllers/main.py

```diff
@@ -990,140 +990,183 @@ def action(self, model, id):
-                image_base64 = res.get(field)
-
-            if kw.get('resize'):
-                resize = kw.get('resize').split(',')
-                if len(resize) == 2 and int(resize[0]) and int(resize[1]):
-                    width = int(resize[0])
-                    height = int(resize[1])
-                    # resize maximum 500*500
-                    if width > 500: width = 500
-                    if height > 500: height = 500
-                    image_base64 = openerp.tools.image_resize_image(base64_source=image_base64, size=(width, height), encoding='base64', filetype='PNG')
-
-            image_data = base64.b64decode(image_base64)
+    # filename
+    if not filename and obj and obj.exists():
+        if hasattr(obj, filename_field):
+            filename = getattr(obj, filename_field)
+        else:
+            filename = "%s-%s-%s" % (obj._model._name, obj.id, field)
+
+    # mintype
+    if download:
+        mimetype = 'application/octet-stream'
+    elif not mimetype:
+        if obj and hasattr(obj, 'mimetype') and obj.mimetype:
+            mimetype = obj.mimetype
+        elif filename:
+            mimetype = mimetypes.guess_type(filename)[0]
+        if not mimetype:
+            mimetype = default_mimetype
+    headers = [('Content-Type', mimetype)]
+
+    # cache
+    etag = hasattr(request, 'httprequest') and request.httprequest.headers.get('If-None-Match')
+    hashed_session = hashlib.md5(request.session_id).hexdigest()
+    retag = hashed_session
+    if etag:
+        if not obj or not id and hashed_session == etag:
+            status = 304
+        elif obj.exists():
+            retag = hashlib.md5(getattr(obj, '__last_update')).hexdigest()
+            if retag == etag:
+                status = 304
+    headers.append(('ETag', retag))
+    if unique:
+        headers.append(('Cache-Control', 'max-age=%s' % STATIC_CACHE))
+    else:
+        headers.append(('Cache-Control', 'max-age=0'))
 
-        except Exception:
-            image_data = self.placeholder()
-        headers.append(('ETag', retag))
-        headers.append(('Content-Length', len(image_data)))
-        try:
-            ncache = int(kw.get('cache'))
-            headers.append(('Cache-Control', 'no-cache' if ncache == 0 else 'max-age=%s' % (ncache)))
-        except:
-            pass
-        return request.make_response(image_data, headers)
+    # content-disposition default name
+    if download:
+        headers.append(('Content-Disposition', content_disposition(filename)))
 
-    def placeholder(self, image='placeholder.png'):
-        addons_path = http.addons_manifest['web']['addons_path']
-        return open(os.path.join(addons_path, 'web', 'static', 'src', 'img', image), 'rb').read()
+    # get content after cache control
+    if status == 200 and (xmlid or id):
+        content = getattr(obj, field)
+        if not content and getattr(obj, 'url', None):
+            return (302, headers, obj.url)
```

Comment:
Nope.
It's wong, specific to some model (it.attachment) and in the wrong place.